docs/aws-china.md

```diff
@@ -0,0 +1,261 @@
+
+Since we are provisioning a cluster in AWS China Region, we need to create a dedicated S3 bucket in AWS China Region.
+
+```bash
+aws s3api create-bucket --bucket prefix-example-com-state-store --create-bucket-configuration LocationConstraint=$AWS_REGION
+```
+
+## [Creating your first cluster](aws.md#creating-your-first-cluster)
+
+### Ensure you have a VPC which can access the internet NORMALLY
+
+First of all, we have to solve the slow and unstable connection to the internet outside China, or the following processes won't work. One way to do that is to build a NAT instance which can route the traffic via some reliable connection. The details won't be discussed here.
+
+### Prepare kops ami
+
+We have to build our own AMI because there is [no official kops ami in AWS China Region][3]. There's two ways to accomplish so. 
+
+#### ImageBuilder
+
+First, launch an instance in a private subnet which accesses the internet fast and stably.
+
+Because the instance launched in a private subnet, we need to ensure it can be connected to the private ip via a vpn or a bastion.
+
+```bash
+SUBNET_ID=<subnet id> # a private subnet
+SECURITY_GROUP_ID=<security group id>
+
+AMI_ID=$(aws ec2 describe-images --filters Name=name,Values=debian-jessie-amd64-hvm-2016-02-20-ebs --query 'Images[*].ImageId' --output text)
+INSTANCE_ID=$(aws ec2 run-instances --image-id $AMI_ID --instance-type m3.medium --key-name $KEY_NAME --security-group-ids $SECURITY_GROUP_ID --subnet-id $SUBNET_ID --no-associate-public-ip-address --query 'Instances[*].InstanceId' --output text)
+aws ec2 create-tags --resources ${INSTANCE_ID} --tags Key=k8s.io/role/imagebuilder,Value=1
+```
+
+Now follow the documentaion of [ImageBuilder][4] in `kube-deploy` to build the image.
+
+```bash
+go get k8s.io/kube-deploy/imagebuilder
+cd ${GOPATH}/src/k8s.io/kube-deploy/imagebuilder
+
+sed -i '' "s|publicIP := aws.StringValue(instance.PublicIpAddress)|publicIP := aws.StringValue(instance.PrivateIpAddress)|" pkg/imagebuilder/aws.go
+make
+${GOPATH}/bin/imagebuilder --config aws.yaml --v=8 --publish false --replicate false --up false
+```
+
+*The execution might fail but the AMI will be created successfully.*
+
+#### Copy AMI from another region
+
+Following [the comment][5] to copy the kops image from another region, e.g. ap-southeast-1.
+
+#### Get the AMI id
+
+No matter how to build the AMI, we get an AMI, e.g. `k8s-1.7-debian-jessie-amd64-hvm-ebs-2017-09-09`.
+
+### [Prepare local environment](aws.md#prepare-local-environment)
+
+Set up a few environment variables.
+
+```bash
+export NAME=myfirstcluster.k8s.local
+export KOPS_STATE_STORE=s3://alo7-prefix-example-com-state-store
+```
+
+### [Create cluster configuration](aws.md#create-cluster-configuration)
+
+We will need to note which availability zones are available to us. AWS China (Beijing) Region only has two availability zones. It will have [the same problem][6], like other regions having less than three AZs, that there is no true HA support.
```

Comment:
No capability to run 3 masters with three azs? We may want to note that we can use three masters in one az.